ebos/eclmpiserializer.hh

```diff
@@ -515,6 +242,174 @@ public:
     }
 
 protected:
+    //! \brief Handler for vectors.
+    //! \tparam T Type for vector elements
+    //! \param data The vector to (de-)serialize
+    template <typename T>
+    void vector(const std::vector<T>& data)
+    {
+        if constexpr (std::is_pod_v<T>) {
+          if (m_op == Operation::PACKSIZE) {
+              (*this)(data.size());
+              m_packSize += Mpi::Packer::packSize(data.data(), data.size(), m_comm);
+          } else if (m_op == Operation::PACK) {
+              (*this)(data.size());
+              Mpi::Packer::pack(data.data(), data.size(), m_buffer, m_position, m_comm);
+          } else if (m_op == Operation::UNPACK) {
+              std::size_t size = 0;
+              (*this)(size);
+              auto& data_mut = const_cast<std::vector<T>&>(data);
+              data_mut.resize(size);
+              Mpi::Packer::unpack(data_mut.data(), size, m_buffer, m_position, m_comm);
+          }
+        } else {
+            if (m_op == Operation::UNPACK) {
+                std::size_t size = 0;
+                (*this)(size);
+                auto& data_mut = const_cast<std::vector<T>&>(data);
+                data_mut.resize(size);
+                std::for_each(data_mut.begin(), data_mut.end(), std::reference_wrapper{*this});
+            } else {
+                (*this)(data.size());
+                std::for_each(data.begin(), data.end(), std::reference_wrapper{*this});
```

Comment:
Tiny nit: You must `#include <functional>` to use `std::reference_wrapper`.  Once you include that header you can also just use the [`std::ref()`](https://en.cppreference.com/w/cpp/utility/functional/ref) helper function to make your intent even more explicit.

Reply:
include added, std:.ref used. i will trigger the new build from upstream.